mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -90,22 +93,35 @@ public void commitTask(TaskAttemptContext originalContext) throws IOException {
     TaskAttemptContext context = TezUtil.enrichContextWithAttemptWrapper(originalContext);
 
     TaskAttemptID attemptID = context.getTaskAttemptID();
-    String fileForCommitLocation = generateFileForCommitLocation(context.getJobConf(),
-        attemptID.getJobID(), attemptID.getTaskID().getId());
-    HiveIcebergRecordWriter writer = HiveIcebergRecordWriter.getWriter(attemptID);
+    JobConf jobConf = context.getJobConf();
+    Map<String, HiveIcebergRecordWriter> writers = HiveIcebergRecordWriter.getWriters(attemptID);
+    Map<String, String> outputs = SerializationUtil.deserializeFromBase64(jobConf.get(InputFormatConfig.OUTPUT_TABLES));
 
-    DataFile[] closedFiles;
-    if (writer != null) {
-      closedFiles = writer.dataFiles();
-    } else {
-      closedFiles = new DataFile[0];
+    ExecutorService tableExecutor = tableExecutor(jobConf, outputs.size());
```

Comment:
Does this need to be done in parallel?

Reply:
See above. Often it does not have to be parallel, and it could be turned off by the user if not needed